src/generic/armcm_boot.c

```diff
@@ -27,12 +28,39 @@ extern uint32_t _stack_end;
 void
 ResetHandler(void)
 {
+    __disable_irq();
+
+    // Explicitly load the stack pointer (for some bootloaders that don't)
+    asm volatile("mov sp, %0\n bx %1"
+                 : : "r"(&_stack_end), "r"(ResetHandlerC));
+}
```

Comment:
It would be preferable to move this to be next to the `DECL_ARMCM_IRQ(ResetHandler, -15);`, and avoid the forward declaration for `ResetHandlerC()`.